junit5/src/main/java/org/jboss/weld/junit5/WeldInitiator.java

```diff
@@ -0,0 +1,172 @@
+     *
+     * @param weld
+     * @return a new test rule
+     */
+    public static WeldInitiator of(Weld weld) {
+        return from(weld).build();
+    }
+
+    /**
+     * The container is configured with the result of {@link #createWeld()} method and all the classes from the test class
+     * package are added.
+     *
+     * @return a new test rule
+     */
+    public static WeldInitiator ofTestPackage() {
+        return fromTestPackage().build();
+    }
+
+    /**
+     * Create a builder instance.
+     *
+     * @param weld
+     * @return a builder instance
+     * @see #of(Class...)
+     */
+    public static Builder from(Class<?>... beanClasses) {
+        return from(createWeld().beanClasses(beanClasses));
+    }
+
+    /**
+     * Create a builder instance.
+     *
+     * @param weld
+     * @return a builder instance
+     * @see #of(Weld)
+     */
+    public static Builder from(Weld weld) {
+        return new Builder(weld);
+    }
+
+    /**
+     * Create a builder instance.
+     *
+     * @return a builder instance
+     * @see #ofTestPackage()
+     */
+    public static Builder fromTestPackage() {
+        return new Builder(null);
+    }
+
+    /**
+     * This builder can be used to customize the final {@link WeldInitiator} instance, e.g. to activate a context for a given
+     * normal scope.
+     */
+    public static final class Builder extends AbstractBuilder {
+
+        public Builder(Weld weld) {
+            super(weld);
+        }
+
+        /**
+         * Activate and deactivate contexts for the given normal scopes for each test method execution.
+         * <p>
+         * {@link ApplicationScoped} is ignored as it is always active.
+         * </p>
+         *
+         * @param normalScopes
+         * @return self
+         */
+        @SafeVarargs
+        public final Builder activate(Class<? extends Annotation>... normalScopes) {
+            return (Builder) super.activate(normalScopes);
+        }
+
+        public Builder inject(Object instance) {
+            return (Builder) super.inject(instance);
+        }
+
+        public Builder addBeans(Bean<?>... beans) {
+            return (Builder) super.addBeans(beans);
+        }
+
+        /**
+         *
+         * @return a new {@link WeldInitiator} instance
+         */
+        @Override
+        public WeldInitiator build() {
+            return new WeldInitiator(weld,
+                instancesToInject.isEmpty() ? Collections.emptyList()
+                    : new ArrayList<>(instancesToInject),
+                scopesToActivate.isEmpty()
+                    ? Collections.<Class<? extends Annotation>>emptySet()
+                    : new HashSet<>(scopesToActivate),
+                beans.isEmpty() ? Collections.<Bean<?>>emptySet() : new HashSet<>(beans));
+        }
+
+    }
+
+    // has to be package protected so we can init this from WeldJunit5Extension is user doesn's specify it
```

Comment:
Following bunch of methods are package protected because the Junit 5 extension needs access to them.
Note that `initWeld` returns `WeldContainer` - we need that in extension to do test method param injection.